api_module.py

```diff
@@ -0,0 +1,319 @@
+
+    :param cursor: The database cursor object to execute SQL commands.
+    :return: A string indicating the bank that have the highest outbound users.
+    """
+    cursor.execute('SELECT Bank_sender_name, Account_sender_id FROM Transact')
+    banks_users_count = {}
+
+    for bank_name, account_id in cursor.fetchall():
+        cursor.execute(SELECT_BY_ID.format('User_id', 'Account'), (account_id,))
+        (user_id) = cursor.fetchone()
+
+        banks_users_count[bank_name] = banks_users_count.get(bank_name, set())
+        banks_users_count[bank_name].add(user_id)
+
+    bank_with_highest_users = max(banks_users_count,
+                                  key=lambda bank: len(banks_users_count[bank]))
+
+    return f'Bank with the highest users which performed outbound transactions ' \
+           f'- {bank_with_highest_users}'
+
+
+def delete_users(cursor):
+    """
+    Delete users with missing or incomplete information from the database.
+
+    :param cursor: A database cursor for executing SQL queries.
+    """
+    cursor.execute('SELECT id FROM User WHERE Name IS NULL OR Surname IS NULL '
+                   'OR Birth_day IS NULL')
+    user_ids_with_missing_info = [row[0] for row in cursor.fetchall()]
+
+    for user_id in user_ids_with_missing_info:
+        delete_user(user_id)
+
+
+def delete_accounts(cursor):
+    """
+    Delete accounts with missing or incomplete information from the database.
+
+    :param cursor: A database cursor for executing SQL queries.
+    """
+    cursor.execute('SELECT id FROM Account WHERE User_id IS NULL OR Type IS NULL '
+                   'OR Account_Number IS NULL OR Bank_id IS NULL OR '
+                   'Currency IS NULL OR Amount IS NULL OR Status IS NULL ')
+    account_ids_with_missing_info = [row[0] for row in cursor.fetchall()]
+
+    for account_id in account_ids_with_missing_info:
+        delete_account(account_id)
+
+
+@db_connection
+def delete_users_and_accounts_with_missing_info(cursor):
+    """
+    Deletes users and accounts from the database if they have missing information.
+
+    :param cursor: The database cursor object to execute SQL commands.
+    :return: A message indicating the success of the operation.
+    """
+    delete_users(cursor)
+    delete_accounts(cursor)
+
+    return 'Users and Accounts with missing information deleted successfully'
+
+
+@db_connection
+def get_user_transactions_last_3_months(cursor, user_full_name):
+    """
+    Retrieve user transactions for last 3 months.
+
+    :param cursor: The database cursor object to execute SQL commands.
+    :param user_full_name: The full name of the user who have the transactions.
+    :return: A list with user transactions.
+    """
+    name, surname = parse_user_full_name(user_full_name)
+    cursor.execute('SELECT id FROM User WHERE Name = ? AND Surname = ?', (name, surname))
+    user_id = cursor.fetchone()
+    check_availability(user_id, 'User not found')
+
+    user_id = user_id[0]
```

Comment:
you already used (user_id) = cursor.fetchone()
or at least user_id = cursor.fetchone()[0]
you don't need an extra line